src/java.base/share/classes/java/util/Locale.java

```diff
@@ -995,11 +995,13 @@ static Locale getInstance(BaseLocale baseloc, LocaleExtensions extensions) {
 
     private static final ReferencedKeyMap<Object, Locale> LOCALE_CACHE = ReferencedKeyMap.create(true, ConcurrentHashMap::new);
     private static Locale createLocale(Object key) {
-        return switch (key) {
-            case BaseLocale base -> new Locale(base, null);
-            case LocaleKey lk -> new Locale(lk.base, lk.exts);
-            default -> throw new InternalError("should not happen");
-        };
+        if (key instanceof BaseLocale base) {
+            return new Locale(base, null);
+        } else if (key instanceof LocaleKey lk) {
+            return new Locale(lk.base, lk.exts);
+        } else {
+            throw new InternalError("should not happen");
```

Comment:
The default branch was required in the switch. Can we simply drop this branch and have a ClassCastException to LocalKey instead?

Reply:
You mean like so:

```
    private static Locale createLocale(Object key) {
        if (key instanceof BaseLocale base) {
            return new Locale(base, null);
        }
        LocaleKey lk = (LocaleKey)key;
        return new Locale(lk.base, lk.exts);
    }
 ```
 ?
 
 I think this should be alright. The type of the "impossibly" thrown exception would differ.

Reply:
Yes, just like how jli handles the Class/MethodType union type arguments.